packages/nuxt/src/core/runtime/nitro/renderer.ts

```diff
@@ -480,21 +478,32 @@ export default defineRenderHandler(async (event): Promise<Partial<RenderResponse
 
   // Response for component islands
   if (isRenderingIsland && islandContext) {
-    const islandHead: NuxtIslandResponse['head'] = {
-      link: [],
-      style: [],
-    }
-    for (const tag of await head.resolveTags()) {
-      if (tag.tag === 'link') {
-        islandHead.link.push({ key: 'island-link-' + hash(tag.props), ...tag.props })
-      } else if (tag.tag === 'style' && tag.innerHTML) {
-        islandHead.style.push({ key: 'island-style-' + hash(tag.innerHTML), innerHTML: tag.innerHTML })
-      }
-    }
+    // re-push inlined styles - head is reset by the island renderer
+    head.push({ style: inlinedStyles })
     const islandResponse: NuxtIslandResponse = {
       id: islandContext.id,
-      head: islandHead,
-      html: getServerComponentHTML(htmlContext.body),
+      head: head.headEntries().filter((h) => {
```

Comment:
if you're just trying to resolve the reactivity, you should use

```ts
import { resolveUnrefHeadInput } from '@unhead/vue'
// ...
for (const entry of ctx.entries)
  entry.resolvedInput = resolveUnrefHeadInput(entry.input)
```

can't we then just serialize this?

Reply:
This is to remove empty arrays from the entries. Do you think we can instead do
`head.headEntries().filter(h => Array.isArray(h) ? h.length : true)` ?

Reply:
Is this just to reduce the payload size? Are empty arrays getting in there through user input or something we're doing :thinking:

Reply:
yes. I'm not sure where it comes from. I'll have to debug that